cupboard_app/views.py

```diff
@@ -166,6 +171,100 @@ def get(self, request: Request) -> Response:
         return Response(serializer.data)
 
 
+@extend_schema(tags=['List_Item'])
+class ListItemAPIView(APIView):
+    missingMsg = "Required value misssing from sent request,"
+    missingMsg += "please ensure all items are sent in the following format:"
+    missingMsg += "{\n  username: [USERNAME],\n  listName: [LISTNAME],\n  ingredient: "
+    missingMsg += "[INGREDIENT],\n  amount: [AMOUNT/QUANTITY],\n  unit: [MEASURMENT UNIT]\n}"
+
+    @extend_schema(
+        request=None,
```

Comment:
Update the request parameter to be a serializer with the request body requirements i.e. username, listName, ingredient, amount, unit.

You can simply create a serializer in the serializers.py by following the **MessageSerializer** format (not the models). For which serializer fields to use, see https://www.django-rest-framework.org/api-guide/fields/